src/jit/assertionprop.cpp

```diff
@@ -1989,11 +1989,23 @@ AssertionInfo Compiler::optAssertionGenJtrue(GenTree* tree)
     op2 = op1->gtCall.gtCallLateArgs->gtOp.gtOp2;
     op1 = op1->gtCall.gtCallLateArgs;
 
+    // For the assertion, ensure op1 is the object being tested.
+    // Morph may have swizzled the operand order.
+    GenTree* op1op = op1->gtOp.gtOp1;
```

Comment:
Wouldn't be simpler/safer to get the calls args from `fgArgInfo`?

Reply:
Ahm, looks like it doesn't work because `optCreateAssertion` is trying to guess that you want a subtype assertion from the fact that it receives a GT_LIST. Oh well...

Reply:
The way assertions are created for helper calls is kind of clunky; the code wants to see the operands wrapped in list nodes.